### PR TITLE
Add test for existing preferred-modules functionality

### PR DESCRIPTION
**Pull request by d33bs**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

The work in this PR adds a simple test for existing preferred-modules functionality.

Background: While exploring solutions for #7957, I noticed there was not yet a specific test for this configuration option. As a result, I opted to add this simple test and decouple the change from new functionality I hope to contribute later towards more fully addressing #7957.

Apologies in advance if I somehow missed the existing work in this area (please don't hesitate to let me know!).

Refs #7957

Thank you for any thoughts, suggestions, or edits you may have as part of the review!

### CODE REVIEW

File: tests/checkers/unittest_imports.py [hunk 1]
```diff
@@ -114,3 +114,26 @@ def test_wildcard_import_non_init(self) -> None:
         )
         with self.assertAddsMessages(msg):
             self.checker.visit_importfrom(import_from)
+
+    @staticmethod
+    def test_preferred_module(capsys: CaptureFixture[str]) -> None:
+        """
+        Tests preferred-module configuration option
+        """
+
```
Comment (DanielNoord) [inline tests/checkers/unittest_imports.py:123]:
```suggestion
```

File: tests/checkers/unittest_imports.py [hunk 2]
```diff
@@ -114,3 +114,26 @@ def test_wildcard_import_non_init(self) -> None:
...
+                "-e preferred-module",
+                # prefer sys instead of os (for triggering test)
+                "--preferred-modules=os:sys",
+            ],
+            exit=False,
+        )
+        output, errors = capsys.readouterr()
+
+        # assert that we saw preferred-modules triggered
+        assert len(output.split("\n")) == 7
```
Comment (DanielNoord) [inline tests/checkers/unittest_imports.py:138]: Could you assert for the message here? If we ever change the output format this will unnecessarily fail.

**Review by DanielNoord — CHANGES_REQUESTED** [review]
Thanks! PRs like these are much appreciated!

File: tests/checkers/unittest_imports.py (hunk 2 above)
Comment (d33bs) [reply tests/checkers/unittest_imports.py]: Thank you for the feedback here! I've made a change to this effect to address this in [4781b0b](https://github.com/PyCQA/pylint/pull/8093/commits/4781b0b96c26865c5a18d6281a01be37c78c95e4).

**Comment by d33bs** [comment]
Thank you @DanielNoord for the review and suggestions! I've made some changes based on what you mentioned. Please don't hesitate to let me know if there is anything else we may address.

**Review by DanielNoord — APPROVED** [review]
Thanks! Love it!

**Pull request merged**